kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemViewKeyInputFormat.java

```diff
@@ -76,24 +77,27 @@ public FileSystemViewKeyInputFormat(FileSystemDataset<E> dataset,
     Format format = dataset.getDescriptor().getFormat();
 
     boolean isSpecific = SpecificRecord.class.isAssignableFrom(dataset.getType());
+    String readerSchema = conf.get("kite.readerSchema");
+    if (isSpecific && readerSchema == null) {
```

Comment:
That was my thought as well. The current implementation doesn't do that. It only handles SpecificRecords and GenericRecords. I think I need to add support for setting the reader schema for ReflectRecords using DataModelUtil.

I can do that in this PR or in a follow-up JIRA.